### ISSUE TITLE
`cmd /c copy` fails with files with plus signs, Can't force quotes around a single argument

### ISSUE THREAD

**Issue by peteristhegreat** [JuliaLang/julia#31343]
Along the lines of https://github.com/JuliaLang/julia/issues/23597#issuecomment-421083998 and supporting cmd built-in somewhat, there is an oddball issue with `cmd /c copy` and plus signs in a filename and quotes.

If I want to copy the file `libstdc++6.dll` on windows, and I use `cmd /c copy libstdc++6.dll libstdc++-7.dll`, it fails because the `+` has special meaning for the `copy` command.  It is used for concatenating two or more files to a new destination.  If I put quotes around it, the `Cmd` or `shell>` strips the quotes before it gets to cmd because the filename doesn't contain spaces.

```
shell> cmd /c dir libstd*
 Volume in drive C has no label.
 Volume Serial Number is 4C01-XXXX

 Directory of C:\Users\username\AppData\Local\Julia-1.0.3\bin

12/17/2018  11:35 PM         1,390,080 libstdc++-6.dll
               1 File(s)      1,390,080 bytes
               0 Dir(s)  155,362,865,152 bytes free

shell> cmd /c copy libstdc++-6.dll libstdc++-7.dll
The system cannot find the file specified.

shell> cmd /c copy "libstdc++-6.dll" "libstdc++-7.dll"
The system cannot find the file specified.

shell> cmd /c copy \"libstdc++-6.dll\" \"libstdc++-7.dll\"
The system cannot find the file specified.
```

```
C:\Users\username>cd C:\Users\username\AppData\Local\Julia-1.0.3\bin\

C:\Users\username\AppData\Local\Julia-1.0.3\bin>dir libstd*
 Volume in drive C has no label.
 Volume Serial Number is 4C01-XXXX

 Directory of C:\Users\username\AppData\Local\Julia-1.0.3\bin

12/17/2018  11:35 PM         1,390,080 libstdc++-6.dll
               1 File(s)      1,390,080 bytes
               0 Dir(s)  155,362,816,000 bytes free

C:\Users\username\AppData\Local\Julia-1.0.3\bin>cmd /c copy "libstdc++-6.dll" "libstdc++-7.dll"
        1 file(s) copied.
```

More testing using `run`
```
julia> run(`cmd /c copy "libstdc++-6.dll" "libstdc++-7.dll"`)
The system cannot find the file specified.
ERROR: failed process: Process(`cmd /c copy libstdc++-6.dll libstdc++-7.dll`, ProcessExited(1)) [1]
Stacktrace:
 [1] error(::String, ::Base.Process, ::String, ::Int64, ::String) at .\error.jl:42
 [2] pipeline_error at .\process.jl:705 [inlined]
 [3] #run#503(::Bool, ::Function, ::Cmd) at .\process.jl:663
 [4] run(::Cmd) at .\process.jl:661
 [5] top-level scope at none:0

julia> copy_cmd = `cmd /c copy "libstdc++-6.dll" "libstdc++-7.dll"`
`cmd /c copy libstdc++-6.dll libstdc++-7.dll`

julia> copy_cmd.exec
5-element Array{String,1}:
 "cmd"
 "/c"
 "copy"
 "libstdc++-6.dll"
 "libstdc++-7.dll"

julia> copy_cmd.exec[4] = "\"libstdc++-6.dll\""
"\"libstdc++-6.dll\""

julia> copy_cmd.exec[5] = "\"libstdc++-7.dll\""
"\"libstdc++-7.dll\""

julia> run(copy_cmd)
The system cannot find the file specified.
ERROR: failed process: Process(`cmd /c copy '"libstdc++-6.dll"' '"libstdc++-7.dll"'`, ProcessExited(1)) [1]
Stacktrace:
 [1] error(::String, ::Base.Process, ::String, ::Int64, ::String) at .\error.jl:42
 [2] pipeline_error at .\process.jl:705 [inlined]
 [3] #run#503(::Bool, ::Function, ::Cmd) at .\process.jl:663
 [4] run(::Cmd) at .\process.jl:661
 [5] top-level scope at none:0
```

Note the last ERROR, where my quoted file names have an additional set of quotes around it, added by Julia.

In summary, for a filename without spaces in it, you can have no quotes or two sets of quotes around it, but not a single set of double quotes, as would be needed for a file with a plus sign in it.

**Comment by StefanKarpinski** [comment]
The first version should work if you’re in the right directory.

**Comment by vtjnash** [comment]
You must set the `windows_verbatim` flag to run the `cmd.exe` binary, and pass a single argument handling escaping on your own

**Comment by StefanKarpinski** [comment]
I'm closing this since Julia is operating correctly here. You may want to ask on [discourse](https://discourse.julialang.org) about how to pass arguments to `cmd.exe` correctly. Note that you can also just use the `cp` command in Julia.

**Comment by peteristhegreat** [comment]
@vtjnash The windows_verbatim flag worked to force quotes.  Thank you.

```
julia> copy_cmd = `cmd /c copy "libstdc++-6.dll" "libstdc++-7.dll"`
`cmd /c copy libstdc++-6.dll libstdc++-7.dll`

julia> copy_cmd2 = Cmd(copy_cmd; windows_verbatim=true)
`cmd /c copy libstdc++-6.dll libstdc++-7.dll`

julia> copy_cmd2.exec[4] = "\"libstdc++-6.dll\""
"\"libstdc++-6.dll\""

julia> copy_cmd2.exec[5] = "\"libstdc++-7.dll\""
"\"libstdc++-7.dll\""

julia> run(copy_cmd2)
        1 file(s) copied.
Process(`cmd /c copy '"libstdc++-6.dll"' '"libstdc++-7.dll"'`, ProcessExited(0))
```
